cupy/core/_cub_reduction.pyx

```diff
@@ -50,9 +50,9 @@ cdef function.Function _create_cub_reduction_function(
 
     cdef str module_code = _get_cub_header_include()
     module_code += '''
+${type_preamble}
 ${preamble}
 
-${type_preamble}
```

Comment:
This is necessary for fusion to work, as it ensures the input/output types are defined before being used in the `preamble` that is emitted by fusion.

Reply:
btw this also follows the same order in CuPy's original implementation (`cupy/core/_reduction.pyx`).